packages/mui-utils/src/deepmerge.ts

```diff
@@ -6,6 +6,20 @@ export interface DeepmergeOptions {
   clone?: boolean;
 }
 
+function clonedeep<T>(source: T): T | Record<keyof any, unknown> {
+  if (!isPlainObject(source)) {
+    return source;
+  }
+
+  const output: Record<keyof any, unknown> = {};
+
+  Object.keys(source).forEach((key) => {
+    output[key] = clonedeep(source[key]);
+  });
+
+  return output;
+}
+
```

Comment:
@oliviertassinari Seems that this implementation fits well, all tests passed and code coverage is happy.